crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -123,22 +152,39 @@ impl<'db> SemanticIndexBuilder<'db> {
         *self
             .scope_stack
             .last()
-            .map(|(scope, _)| scope)
+            .map(|ScopeInfo { file_scope_id, .. }| file_scope_id)
             .expect("Always to have a root scope")
     }
 
     fn loop_state(&self) -> LoopState {
         self.scope_stack
             .last()
             .expect("Always to have a root scope")
-            .1
+            .loop_state
+    }
+
+    /// Returns the scope ID of the surrounding class body scope if the current scope
+    /// is a method inside a class body. Returns `None` otherwise, e.g. if the current
+    /// scope is a function body outside of a class, or if the current scope is not a
+    /// function body.
+    fn is_method_of_class(&self) -> Option<FileScopeId> {
```

Comment:
What about cases where the method is conditionally defined by using an if statement?

Reply:
`if` statements do not introduce new scopes, so this works fine (see also the "Conditionally declared / bound attributes" test)